### PR TITLE
Publish message via send task

### LINKED ISSUE #13398: I can publish a message via send task
**Is your feature request related to a problem? Please describe.**
As a user, i want to model a process where a `Send Task` sends a message.

**Additional context**
```xml
<bpmn:sendTask id="Activity_1kgyl85" messageRef="Message_1nb8aa6">
    <bpmn:extensionElements>
       <zeebe:publishMessage correlationKey="= some_other_order_id" />
    </bpmn:extensionElements>
</bpmn:sendTask>

<bpmn:message id="Message_1nb8aa6" name="order_placed" />
```

### PR DESCRIPTION
**Pull request by lzgabel**
## Description
As a user, I can publish a message via `Send Task`.

## Related issues
closes #13398

## Definition of Done

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by lzgabel** [comment]
Hi @korthout. Please take a moment to review this feature. Thanks! :heart:

File: engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnMessageBehavior.java
Comment (korthout) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnMessageBehavior.java:127]: ❌ The message must be published on a specific partition, and without a key.

The partition id is selected by hashing the `correlationKey` to the partition where a subscription for this specific message can be opened.

💡 You can use `SubscriptionUtil.getSubscriptionPartitionId` to find the correct partition id.

https://github.com/camunda/zeebe/blob/1c84f9694d7be612f6d346e436d1c6c25ab160dc/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/SubscriptionUtil.java#L32-L44

💡 For sending the message to the selected partition, you can use `InterPartitionCommandSender.sendCommand` (choose the option that does not take a `recordKey`).

https://github.com/camunda/zeebe/blob/1c84f9694d7be612f6d346e436d1c6c25ab160dc/stream-platform/src/main/java/io/camunda/zeebe/stream/api/InterPartitionCommandSender.java#L14-L23

**Review by korthout — CHANGES_REQUESTED** [review]
Thanks @lzgabel This is really amazing work! 👏 

❌ There are a few issues that must be resolved before we can merge this. Please have a look.

**Comment by lzgabel** [comment]
Hi @korthout. Please check this again. :bow:

BTW. I'll squash all commits when you're done checking this out. :heart:

**Review by korthout — CHANGES_REQUESTED** [review]
Thanks again for these changes @lzgabel 👏 

🙇 This time, I have some bad news because I overlooked something major. Publishing the command to the other partition is unreliable, and we have no quick solutions to avoid this.

✍️ We need to return to the drawing board to design how we can make this reliable and find a way to respond to the successful publishing of the message. @remcowesterhoud and I have some ideas for this, building upon #11456, but it may take time to work out the details.

⏸️ Therefore, I propose to put this pull request on hold until further notice. I hope you understand.

🥇 I want to thank you once more for your contributions on this topic. You've done an absolutely incredible job.

❓ Lastly, would you be willing to continue #13241 while this is on hold? That topic is defined in detail, and we should not expect such a similar problem as #11456 was specifically designed with signals in mind.

**Comment by lzgabel** [comment]
> Lastly, would you be willing to continue https://github.com/camunda/zeebe/issues/13241 while this is on hold? That topic is defined in detail, and we should not expect such a similar problem as https://github.com/camunda/zeebe/issues/11456 was specifically designed with signals in mind.

@korthout. 🤝  Ok, let's put this topic on hold for now , I will continue #13241.

**Comment by Zelldon** [comment]
I will close this PR (currently cleaning stale PRs) this seems to be stale for a while and has already some merge conflicts. Feel free to reopen it if it becomes up to data again.

**Pull request closed without merging**
